source/extensions/filters/udp/dns_filter/dns_parser.h

```diff
@@ -103,7 +99,29 @@ class DnsAnswerRecord : public BaseDnsRecord {
 };
 
 using DnsAnswerRecordPtr = std::unique_ptr<DnsAnswerRecord>;
-using DnsAnswerMap = absl::flat_hash_map<std::string, std::list<DnsAnswerRecordPtr>>;
+using DnsAnswerMap = std::unordered_multimap<std::string, DnsAnswerRecordPtr>;
+
+/**
+ * DnsQueryContext contains all the data associated with a query. The filter uses this object to
+ * generate a response and determine where it should be transmitted.
+ */
+class DnsQueryContext {
+public:
+  DnsQueryContext(Network::Address::InstanceConstSharedPtr local,
+                  Network::Address::InstanceConstSharedPtr peer)
+      : local_(std::move(local)), peer_(std::move(peer)), status_(false), id_() {}
+  ~DnsQueryContext() = default;
+
+  Network::Address::InstanceConstSharedPtr local_;
+  Network::Address::InstanceConstSharedPtr peer_;
+
+  bool status_;
```

Comment:
status_  didn't quite tell me what it was until I browsed the .cpp to see that it means parsing the request failed. Can we come up with a more self-explanatory name for this?

Reply:
Yes, I have this renamed locally, I'll update it.